loader/src/reference_binder.py

```diff
@@ -223,3 +168,44 @@ def load_cached_references(self, target_ref: str) -> DefaultDict[tuple, list]:
             (source_ref, ref) = json.loads(element)
             pending_refs[tuple(source_ref)].append(ref)
         return pending_refs
+
+    @staticmethod
+    def identifier_to_key(resource_type, identifier):
+        value = identifier.get("value")
+        system = identifier.get("system")
+        return f"{resource_type}:{value}:{system}"
+
+    @staticmethod
+    def partial_identifier(identifier):
+        return {
+            "identifier.value": identifier.get("value"),
+            "identifier.system": identifier.get("system")
+        }
+
+    def build_find_predicate(self, resource_ids, reference_path, identifier, is_array):
+        """Finds all resources with unresolved references to a given identifier"""
+        res = {
+            "id": {
+                "$in": resource_ids
+            }
+        }
+        if is_array:
+            res[reference_path] = {
+                "$elemMatch": self.partial_identifier(identifier)
+            }
+        else:
+            res[f"{reference_path}.identifier.value"] = identifier.get("value")
+            res[f"{reference_path}.identifier.system"] = identifier.get("system")
```

Comment:
I know it was there before but I don't get it: why do we need more than the ids for the find predicate?

Reply:
No reason I can think of. Removed.